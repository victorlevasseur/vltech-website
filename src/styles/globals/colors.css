:root {
  --color-black: #212427;
  --color-white: #ffffff;
  --color-transparent: #ffffff00;
  
  /* Primary colors */
  --color-primary-50: #e8f3ff;
  --color-primary-100: #b9daff;
  --color-primary-200: #b7c9ff;
  --color-primary-300: #67b0ff;
  --color-primary-400: #49a1ff;
  --color-primary-500: #1c89ff;
  --color-primary-600: #197de8;
  --color-primary-700: #1461b5;
  --color-primary-800: #0f4b8c;
  --color-primary-900: #0c3a6b;

  /* Secondary colors */
  --color-secondary-50: #efe6fd;
  --color-secondary-100: #ceb0fa;
  --color-secondary-200: #b78af7;
  --color-secondary-300: #9654f4;
  --color-secondary-400: #8133f1;
  --color-secondary-500: #6200ee;
  --color-secondary-600: #5900d9;
  --color-secondary-700: #4600a9;
  --color-secondary-800: #360083;
  --color-secondary-900: #290064;

  /* Neutral colors */
  --color-neutral-50: #f3f3f3;
  --color-neutral-100: #dadada;
  --color-neutral-200: #c8c8c8;
  --color-neutral-300: #afafaf;
  --color-neutral-400: #a0a0a0;
  --color-neutral-500: #888888;
  --color-neutral-600: #7c7c7c;
  --color-neutral-700: #616161;
  --color-neutral-800: #4b4b4b;
  --color-neutral-900: #393939; 
}