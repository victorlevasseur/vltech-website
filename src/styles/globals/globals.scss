@use '../tokens/colors.scss' as colors;
@use '../tokens/font.scss' as font;
@use '../tokens/media.scss' as media;
@use '../tokens/typo.scss' as typo;

@import 'modern-normalize/modern-normalize.css';

@import 'colors.css';

* {
  @include font.normal;
  
  color: colors.black();
}

html, body {
  margin: 0;
  height: 100%;
}

p, ul, ol {
  @include typo.normal;

  margin: 16px 0;
}

a {
  @include typo.link;

  color: colors.primary(500);

  &:hover {
    color: colors.primary(700);
    text-decoration: underline;
    text-decoration-thickness: 2px;
    text-underline-offset: 0.1em;
  }
}

em {
  font-style: normal;
  font-weight: 600;
}

h1 {
  @include typo.heading-1;
}

h2 {
  @include typo.heading-2;
}

h3 {
  @include typo.heading-3;
}

@each $breakpoint, $size in media.$breakpoints {
  .hide-smaller-#{$breakpoint} {
    @include media.smaller($breakpoint) {
      display: none;
    }
  }

  .hide-larger-or-eq-#{$breakpoint} {
    @include media.larger-or-eq($breakpoint) {
      display: none;
    }
  }
}