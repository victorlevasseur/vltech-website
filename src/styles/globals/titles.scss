@use '../tokens/clamp.scss';
@use '../tokens/colors.scss';
@use '../tokens/font.scss';
@use '../tokens/typo.scss';


@mixin titles {
  h1 {
    @include typo.heading-1;
  }

  h2 {
    @include typo.heading-2;
  }

  h3 {
    @include typo.heading-3;
  }

  hgroup {
    h1, h2, h3 {
      margin-bottom: 4px;
    }

    &:has(h1) {
      margin-bottom: 18px;
    }

    &:has(h2) {
      margin-bottom: 12px;
    }

    &:has(h3) {
      margin-bottom: 18px;
    }

    p {
      margin: 4px 0;
      border-left: 5px solid colors.primary(500);
      padding: 2px 0 2px 6px;

      @include font.catchy;
      font-size: clamp.slope(14px, 16px);
      font-weight: 600;
    }
  }
}