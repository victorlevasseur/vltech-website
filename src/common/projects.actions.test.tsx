import { expect } from '@jest/globals';
import {
  listProjectIds,
  listProjects,
  readProject,
} from '@/common/projects.actions';
import { listFilesInFolder } from '@/common/fs.actions';

jest.mock('../data/technology/data');

jest.mock<typeof import('./fs.actions')>('./fs.actions', () => ({
  listFilesInFolder: jest.fn(() => Promise.resolve(['file1', 'file2'])),
}));

describe('readProject', () => {
  test('should return a parsed Project interface', async () => {
    // Fake MDX doc.
    jest.doMock(
      '@/data/projects/contents/project.mdx',
      () => ({
        __esModule: true,
        default: () => (
          <div>
            This is a <strong>test</strong>.
          </div>
        ),
        frontmatter: {
          name: 'Project',
          summary: 'project summary',
          technologies: ['victorscript', 'arthurscript'],
        },
      }),
      { virtual: true }
    );

    const project = await readProject('project');

    expect(project).toEqual({
      id: 'project',
      name: 'Project',
      summary: 'project summary',
      content: expect.anything(),
      technologies: [
        {
          category: {
            id: 'CategoryB',
            name: 'Category B',
            priority: 0,
          },
          id: 'victorscript',
          name: 'VictorScript',
        },
        {
          category: {
            id: 'CategoryB',
            name: 'Category B',
            priority: 0,
          },
          id: 'arthurscript',
          name: 'ArthurScript',
        },
      ],
    });

    expect(project.content({})).toMatchSnapshot();
  });

  test('should fail if name metadata is missing', async () => {
    jest.doMock(
      '@/data/projects/contents/project-name-missing.mdx',
      () => ({
        __esModule: true,
        default: () => (
          <div>
            This is a <strong>test</strong>.
          </div>
        ),
        frontmatter: {
          summary: 'project summary',
          technologies: ['victorscript', 'arthurscript'],
        },
      }),
      { virtual: true }
    );

    await expect(async () => {
      await readProject('project-name-missing');
    }).rejects.toThrow('Project name is missing!');
  });

  test('should fail if summary metadata is missing', async () => {
    jest.doMock(
      '@/data/projects/contents/project-summary-missing.mdx',
      () => ({
        __esModule: true,
        default: () => (
          <div>
            This is a <strong>test</strong>.
          </div>
        ),
        frontmatter: {
          name: 'Project',
          technologies: ['victorscript', 'arthurscript'],
        },
      }),
      { virtual: true }
    );

    await expect(async () => {
      await readProject('project-summary-missing');
    }).rejects.toThrow('Project summary is missing!');
  });

  test('should fail if technologies metadata is missing', async () => {
    jest.doMock(
      '@/data/projects/contents/project-technologies-missing.mdx',
      () => ({
        __esModule: true,
        default: () => (
          <div>
            This is a <strong>test</strong>.
          </div>
        ),
        frontmatter: {
          name: 'Project',
          summary: 'project summary',
        },
      }),
      { virtual: true }
    );

    await expect(async () => {
      await readProject('project-technologies-missing');
    }).rejects.toThrow('Project technologies are missing!');
  });
});

describe('listProjectIds', () => {
  test('should return a list of projects ids', async () => {
    expect(await listProjectIds()).toEqual(['file1', 'file2']);

    expect(listFilesInFolder).toHaveBeenCalledWith(
      'src/data/projects/contents',
      'mdx'
    );
  });
});

describe('listProjects', () => {
  test('should return all projects listed and parsed', async () => {
    jest.doMock(
      '@/data/projects/contents/file1.mdx',
      () => ({
        __esModule: true,
        default: () => (
          <div>
            This is a <strong>test</strong>.
          </div>
        ),
        frontmatter: {
          name: 'File 1',
          summary: 'file 1 summary',
          technologies: ['victorscript', 'arthurscript'],
        },
      }),
      { virtual: true }
    );

    jest.doMock(
      '@/data/projects/contents/file2.mdx',
      () => ({
        __esModule: true,
        default: () => (
          <div>
            This is another <strong>test</strong>.
          </div>
        ),
        frontmatter: {
          name: 'File 2',
          summary: 'file 2 summary',
          technologies: ['victorscript', 'totorCloud'],
        },
      }),
      { virtual: true }
    );

    const projects = await listProjects();

    expect(projects).toEqual([
      {
        id: 'file1',
        name: 'File 1',
        summary: 'file 1 summary',
        content: expect.anything(),
        technologies: [
          {
            category: {
              id: 'CategoryB',
              name: 'Category B',
              priority: 0,
            },
            id: 'victorscript',
            name: 'VictorScript',
          },
          {
            category: {
              id: 'CategoryB',
              name: 'Category B',
              priority: 0,
            },
            id: 'arthurscript',
            name: 'ArthurScript',
          },
        ],
      },
      {
        id: 'file2',
        name: 'File 2',
        summary: 'file 2 summary',
        content: expect.anything(),
        technologies: [
          {
            category: {
              id: 'CategoryB',
              name: 'Category B',
              priority: 0,
            },
            id: 'victorscript',
            name: 'VictorScript',
          },
          {
            category: {
              id: 'CategoryA',
              name: 'Category A',
              priority: 1,
            },
            id: 'totorCloud',
            name: 'Totor Cloud',
          },
        ],
      },
    ]);

    expect(projects[0].content({})).toMatchSnapshot();
    expect(projects[1].content({})).toMatchSnapshot();
  });
});
