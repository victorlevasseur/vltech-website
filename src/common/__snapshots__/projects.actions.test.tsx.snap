// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listProjects should return all projects listed and parsed 1`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "children": [
      "This is a ",
      {
        "$$typeof": Symbol(react.transitional.element),
        "_owner": null,
        "_store": {},
        "key": null,
        "props": {
          "children": "test",
        },
        "type": "strong",
      },
      ".",
    ],
  },
  "type": "div",
}
`;

exports[`listProjects should return all projects listed and parsed 2`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "children": [
      "This is another ",
      {
        "$$typeof": Symbol(react.transitional.element),
        "_owner": null,
        "_store": {},
        "key": null,
        "props": {
          "children": "test",
        },
        "type": "strong",
      },
      ".",
    ],
  },
  "type": "div",
}
`;

exports[`readProject should return a parsed Project interface 1`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "children": [
      "This is a ",
      {
        "$$typeof": Symbol(react.transitional.element),
        "_owner": null,
        "_store": {},
        "key": null,
        "props": {
          "children": "test",
        },
        "type": "strong",
      },
      ".",
    ],
  },
  "type": "div",
}
`;
